### ISSUE TITLE
Make ModifiedControlVariable detect forEach loop

### ISSUE THREAD

**Issue by Zustin** [checkstyle/checkstyle#11033]
I have read check documentation: https://checkstyle.sourceforge.io/config_coding.html#NestedForDepth
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run

```bash
/var/tmp $ javac Test.java
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="ModifiedControlVariable"/>
  </module>
</module>

/var/tmp $ cat Test.java
import java.util.*;
public class Test {
    public void foo() {
        ArrayList<Integer> list = new ArrayList<>();
        list.add(1);
        list.add(2);
        list.forEach((i) -> {
            i *= 10;  // should report a warning
        });
        // for(Integer i : list) {
        //     i *= 10;  // can be detected
        // }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

---
**Describe what you expect in detail.**
---
Hi, I found a false negative caused by new java characteristic. Checkstyle should report a warning at line 8, but it doesn't. However, the code commented below can be detected and they care equivalent. Thanks.

**Comment by Vyom-Yadav** [comment]
@Zustin Thanks for the report, just a minor, link to documentation is not correct, should be https://checkstyle.sourceforge.io/config_coding.html#ModifiedControlVariable

**Comment by Zustin** [comment]
@Vyom-Yadav Hi, thanks for your efficient response. I find this rule is also affected by identical cause and submit a new issue #11035 . Thanks for your kind consideration.

**Comment by romani** [comment]
from documentation:
> Checks that for loop control variables are not modified inside the for block. 

Check is targeting only `for` block variables.

you need to use https://checkstyle.org/config_coding.html#ParameterAssignment to catch such assignments.
but you found a bug in such Check, newly created issue https://github.com/checkstyle/checkstyle/issues/11038
